File: C/Exercice4_3_fils.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main(int argc, char *argv[])
{
    int tube, retour;
    char buffer[256];

    if(argc < 2)
    {
        puts("Erreur d'argument\n");
        retour = 1;
    }
    else
    {
        tube = atoi(argv[1]);
        read(tube, buffer, sizeof(buffer));

        printf("Le message du pere: %s\n", buffer);
        close(tube);
        retour = 0;
    }

    return retour;
}

File: Python/Exercice4_3_fils.py
#coding: utf-8
import sys
import os

if len(sys.argv) < 2:
    print("Erreur d'argument\n")
    sys.exit(1)

tube = int(sys.argv[1])
buffer = os.read(tube, 256).decode()
print(f"Le message du pere: {buffer}")
os.close(tube)

sys.exit(0)